src/Umbraco.Web/GridFrameworks/UmbracoGridFramework.cs

```diff
@@ -0,0 +1,171 @@
+        {
+            Mandate.That<ArgumentNullException>(HasGridValue);
+            var sectionsDiv = GridHelper.GetDivWrapper(RowCssClass);
+            sectionsDiv.AddClassName("clearfix");
+
+            foreach (var columHtml in GridValue.Sections.Select(section => GetSectionHtml(html, GridValue.Sections.IndexOf(section), beforeRowRender)))
+            {
+                var sectionDiv = GridHelper.GetDivWrapper(SectionCssClass);
+                sectionDiv.AddChild(columHtml.ToHtmlString());
+                sectionsDiv.AddChild(sectionDiv);
+            }
+
+            var containerDiv = GridHelper.GetDivWrapper(ContainerCssClass);
+            return containerDiv.AddChild(sectionsDiv).ToHtml();
+        }
+
+        /// <inheritdoc />
+        public IHtmlString GetRowHtml(HtmlHelper html, GridValue.GridRow row, bool wrapInContainer, Func<HtmlTagWrapper, HtmlTagWrapper> beforeRowRender)
+        {
+            Mandate.That<ArgumentNullException>(row != null);
+            var rowDiv = GridHelper.GetDivWrapper(RowCssClass);
+            rowDiv.AddClassName("clearfix");
+
+            foreach (var area in row.Areas)
+            {
+                rowDiv.AddChild(GetAreaHtml(html, area).ToHtmlString());
+            }
+
+            var rowDivWrapper = GridHelper.GetDivWrapper(row);
+            rowDivWrapper.AddChild(
+                wrapInContainer
+                    ? GridHelper.GetDivWrapper(ContainerCssClass).AddChild(rowDiv)
+                    : rowDiv
+            );
+            return beforeRowRender != null
+                ? beforeRowRender(rowDivWrapper).ToHtml()
+                : rowDivWrapper.ToHtml();
+        }
+
+        /// <inheritdoc />
+        public IHtmlString GetAreaHtml(HtmlHelper html, GridValue.GridArea area)
+        {
+            Mandate.That<ArgumentNullException>(area != null);
+            var areaDivWrapper = GridHelper.GetDivWrapper(area);
+
+            // Render all the controls to HtmlStrings
+            foreach (var control in area.Controls.Where(c => c != null && c.Editor != null && c.Editor.View != null))
+            {
+                areaDivWrapper.AddChild(GetControlHtml(html, control).ToHtmlString());
+            }
+
+            var areaColumnDiv = GridHelper.GetDivWrapper(ColumnCssClassPreText + area.Grid);
+            areaColumnDiv.AddClassName(ColumnCssClass);
+            areaColumnDiv.AddChild(areaDivWrapper);
+            return areaColumnDiv.ToHtml();
+        }
+
+        /// <inheritdoc />
+        public virtual IHtmlString GetControlHtml(HtmlHelper html, GridValue.GridControl control)
+        {
+            var view = (control.Editor.Render ?? control.Editor.View).Replace(".html", ".cshtml");
+            if (view.Contains("/") == false)
+            {
+                view = PartialViewsPath + view;
+            }
+            return html.Partial(view, JToken.FromObject(control));
+        }
```

Comment:
# @Shazwazza This where GridControl is converted to a JToken and passed to the views accepting dynamics

Reply:
@Shazwazza  Currently the GridControl object is converted to a JToken and passed to the view the model.  The grid editor views could be changed to accept a GridControl as the model instead of a dynamic.  With the integration of NestedContent, an alternative would be to not use a separate GridControl object but allow an IPublishedContent instead.  This would require changing the Grid datatype to accept document types.  DocTypeGridEditor sort of allows this, but I'd like to see document types as first class citizens instead.